apps/lib/app_provider.c

```diff
@@ -44,6 +46,8 @@ static int opt_provider_load(const char *provider)
 static void provider_free(OSSL_PROVIDER *prov)
 {
     OSSL_PROVIDER_unload(prov);
+    OPENSSL_CTX_free(libctx);
+    libctx = NULL;
```

Comment:
These two lines should be in `app_providers_cleanup()`

Reply:
removed this file - its in another bigger PR.